File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/RecoveryApi.swift
import Foundation

/// Facade for identity recovery operations.
public struct RecoveryApi {
    private let manager: RecoveryManager

    init(manager: RecoveryManager) {
        self.manager = manager
    }

    public func generateRecoveryKey(identity: Identity) async throws -> Data {
        try await manager.generateRecoveryKey(identity: identity)
    }

    public func hasRecoveryKey(keyId: String) async -> Bool {
        await manager.hasRecoveryKey(keyId: keyId)
    }

    public func restoreWithRecoveryKey(
        did: String,
        recoveryPrivateKeyBase64: String,
        name: String,
        algorithm: Algorithm
    ) async throws -> Identity {
        try await manager.restoreWithRecoveryKey(
            did: did,
            recoveryPrivateKeyBase64: recoveryPrivateKeyBase64,
            name: name,
            algorithm: algorithm
        )
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/RecoveryApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.Algorithm
import my.ssdid.sdk.domain.model.Identity
import my.ssdid.sdk.domain.recovery.RecoveryManager

class RecoveryApi internal constructor(private val manager: RecoveryManager) {
    suspend fun generateRecoveryKey(identity: Identity): Result<ByteArray> =
        manager.generateRecoveryKey(identity)
    suspend fun hasRecoveryKey(keyId: String): Boolean =
        manager.hasRecoveryKey(keyId)
    suspend fun restoreWithRecoveryKey(
        did: String,
        recoveryPrivateKeyBase64: String,
        name: String,
        algorithm: Algorithm
    ): Result<Identity> =
        manager.restoreWithRecoveryKey(did, recoveryPrivateKeyBase64, name, algorithm)
}
